lib/ClangImporter/ImportMacro.cpp

```diff
@@ -779,10 +779,11 @@ ValueDecl *ClangImporter::Implementation::importMacro(Identifier name,
   PrettyStackTraceStringAction stackRAII{"importing macro", name.str()};
 
   // Look for macros imported with the same name.
-  auto known = ImportedMacros.find(name);
-  if (known == ImportedMacros.end()) {
+  auto [known, inserted] = ImportedMacros.try_emplace(
+      name, SmallVector<std::pair<const clang::MacroInfo *, ValueDecl *>, 2>{});
```

Comment:
Can we use something like `auto` here to avoid repeating the type name?

Reply:
Unfortunately, I could not avoid naming the type here because the value for `try_emplace` is taking a template so there is no context to deduce the type of this `SmallVector` from.